server/controllers/Profile.js

```diff
@@ -23,4 +23,44 @@ const getProfile = async (req, res) => {
     }
 }
 
-module.exports = {getProfile};
\ No newline at end of file
+const addAllergy = async (req, res) => {
+    try {
+        const {id, UserId, ingredient} = req.body;
+        if (!ingredient) {
+            res.status(400);
+            return res.json({ error: "Field(s) left empty" });
+        }
+
+        if (!UserId) {
+            res.status(400);
+            return res.json({ error: "No userid provided" });
+        }
+
+        const profile = await Profile.findOne({
```

Comment:
Why are you searching based on profile id instead of searching for the userId that would map to a induvial user?